api/src/model/request/play_cellular_data_confirmation_request.gd

```diff
@@ -0,0 +1,74 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# Request object that handles asynchronous logic related to 
+# show_cellular_data_confirmation().
+#
+# Emits request_completed signal upon success/error.
+# The first boolean argument will be true and the second argument will contain 
+# an integer value, represented by an AssetPackStorageMethod enum if remove 
+# request succeeds. Otherwise the second argument will contain a 
+# PlayAssetPackException object representing the exception encountered.
+# 
+# This object also provides relevant getters so that it is possible to retrieve
+# the updated states from this object using the yield to signal approach.
+#
+# ##############################################################################
+class_name PlayCellularDataConfirmationRequest
+extends PlayAssetDeliveryRequest
+
+signal request_completed
+
+var _status : bool
+var _result : int
+var _error : PlayAssetPackException
+
+# -----------------------------------------------------------------------------
+# Returns boolean indicating Request succeeded/failed.
+# -----------------------------------------------------------------------------
+func get_status() -> bool:
+	return _status
+
+# -----------------------------------------------------------------------------
+# Returns the result of a succeeded Request, represent by an 
+# AssetPackStorageMethod enum.
+# -----------------------------------------------------------------------------
+func get_result() -> int:
+	return _result
+
+# -----------------------------------------------------------------------------
+# Returns a PlayAssetPackException if Request failed.
+# -----------------------------------------------------------------------------
+func get_error() -> PlayAssetPackException:
+	return _error
+
+# -----------------------------------------------------------------------------
+# Callback functions handling signals emitted from the plugin.
+# -----------------------------------------------------------------------------
+func on_show_cellular_data_confirmation_success(result : int):
+	_status = true
+	_result = result
+	emit_signal("request_completed", true, result)
```

Comment:
I think we discussed this request_{completed, succeeded, failed} decision during the design doc meeting and removing the request_completed signal will make it hard for devs who are yielding on that signal.

I think refactoring this signal to take three arguments is definitely a good call. I added this change in the most recent commit.
Since for this result the type is int we are unable to return null. Is returning -1 for result a good decision here?

Reply:
Fixed a bug where the wrong enum is used, create a new 
```
enum CellularDataConfirmationResult {
	RESULT_UNDEFINED = -2,
	RESULT_OK = -1,
	RESULT_CANCELED = 0
}
```
to handle this situation.